### DISCUSSION TITLE
Dreamehome Account Support

### DISCUSSION THREAD

**Discussion by avi-ts** [Tasshack/dreame-vacuum#109] [Ideas]
The new app for the robots of dreambot
She "Dreamehome"
Is it possible to make a choice in the integration according to which application to connect?
The new application has new options and it is better to use it and not through the Xiaomi app

**Comment by Tasshack** [comment]
Integration only works with Mi Home app.
https://github.com/Tasshack/dreame-vacuum/issues/39

**Comment by NilsMinor** [comment]
Is it planed to support this in the future? :)

**Comment by MichelFR** [comment]
Im willing to help implementing this by reverse engineering the Dreamehome app. I'll look into it.

**Reply by avi-ts** [reply]
At first glance the app is very similar, maybe even almost the same, it's worth checking maybe all the difference is in the servers

**Comment by Tasshack** [comment]
Dreamehome app uses [Alibaba Cloud](https://www.alibabacloud.com/) instead of Xiaomi servers. I have checked the api documentation and it seems like private token is required to communicate with the api. If that was the case, it is impossible to get a 3rd party implemention working properly or legally.

**Comment by hiddevanbrussel** [comment]
If its possible it would be nice.

**Comment by TRusselo** [comment]
I got my Dreame today. installed Dreame app,  found out that integration needs Mi Home app.
So.. I Changed apps.
it took 60 seconds to transfer my machine to new app.  
it KEPT my maps and settings from Dreame app and pulled all into mi home app.  i lost nothing.

Just change apps! its so easy you will feel dumb for not doing it sooner.

**Reply by avi-ts** [reply]
Only in the official Dream app can you access the robot's camera or has it already been updated?

**Reply by feerlessleadr** [reply]
@TRusselo Question for you - How did you transfer your vacuum/maps over from the Dreame app to the Mi Home app? I amd trying to do the same thing but it seems I need to reset the vacuum first, and I'm not sure if that will make me lose my maps or not. Thanks

**Reply by Tasshack** [reply]
@feerlessleadr https://github.com/Tasshack/dreame-vacuum/issues/209#issuecomment-1707217346

**Reply by Reudo** [reply]
After the reset and the new connection to the dreame app my map got restored in the new app.

**Reply by feerlessleadr** [reply]
@Tasshack thanks, I reset the vacuum and my maps were restored automatically. I'm going to play around with the integration after my kids are asleep.

**Comment by Tasshack** [comment]
@MichelFR are you still interested on adding the dreamehome cloud feature to this integration because I have done almost half of the work. Looks like the plugin is almost the same with the mi home one but the cloud api burried in a native static library of the dreamehome app. I have even decompiled all the static libraries but i clouldn't find the dreamehome api sdk sources.
I have extracted the private api token from the app and be able to login and get the device list from the dreame cloud services.
Looks like api parameters are the same but endpoints are completely different from the mi cloud.
So far I have found the endpoint for getting and setting the device properties.
`https://cn.iot.dreame.tech:13267/dreame-iot-com-10000/device/sendCommand`
Getting the device list
`https://cn.iot.dreame.tech:13267/dreame-user-iot/iotuserbind/device/listV2`
Login
`https://cn.iot.dreame.tech:13267/dreame-auth/oauth/token`

I need to find rest of the endpoints but I couldn't be able to intercept https traffic from the new version of the app and the older version does not let me use the device because it's firmware is newer. Latest version of the app even doesn't work on rooted devices.

I need the equavilent of these endpoints for the dreamehome. 
```
https://api.io.mi.com/app/v2/home/rpc/{did} (for making local api requests over cloud)
https://api.io.mi.com/app/user/get_user_device_data (for getting the map data object name)
https://api.io.mi.com/app/home/getfileurl (for getting the cleaning history map data)
https://api.io.mi.com/app/v2/home/get_interim_file_url (for getting the map data)
https://api.io.mi.com/app/device/batchdevicedatas (for getting the cloud user settings)
```

I need someone to help me intercept https traffic from mobile app (ios or android) to the dreame cloud servers.

**Reply by MichelFR** [reply]
To be honest, I tried to reverse engineer the app, but failed. I must pass on this one.

But I'm happy to hear that you made some progress, I would happily donate some bucks to get this functionality running.

**Comment by Tasshack** [comment]
I have managed to decompile the app and successfuly extract the sdk sources. 
Dreamehome account login feature will be here soon.

**Reply by avi-ts** [reply]
Wonderful news!  Well done!

**Reply by MichelFR** [reply]
Great news, thanks for your work! 👍🏽

**Reply by drashish87** [reply]
Great news, many thanks for adding support for Dreamehome app!

**Reply by cobra1037** [reply]
you are my hero :) thank you in advance!

**Comment by Tasshack** [comment]
Reopened a related issue about this https://github.com/Tasshack/dreame-vacuum/issues/39

**Comment by Drealine** [comment]
@Tasshack , I've got a vacuum support only Dreame account. If you've interested to help you, I can. 
Do you've ETA for this feature ?

**Comment by Tasshack** [comment]
https://github.com/Tasshack/dreame-vacuum/releases/tag/v2.0.0b6

**Comment by RobertWojtowicz** [comment]
Hi,

@Tasshack

It works with D10s Pro! is it possible to completely disconnect from the cloud after pulling the configuration to HA ?
Is there a chance to stream to HA from D10s Pro camera ?
I know it's a beta version, but you can't see, for example, accessory usage statuses.

Br,
Robert

**Comment by Tasshack** [comment]
@RobertWojtowicz 
> It works with D10s Pro! is it possible to completely disconnect from the cloud after pulling the configuration to HA ?

Cloud connection is required with the Dreamehome account.

> Is there a chance to stream to HA from D10s Pro camera ?

Camera streaming is not possible yet.

> I know it's a beta version, but you can't see, for example, accessory usage statuses.

Consumable entities are already implemented but disabled by default, you just need to enable the entities from device page.

**Comment by haraldbootz** [comment]
EDIT: solved. Found out about enabling beta in HACS, downloading the beta (now at v2.0.0b11) and everything is working. Thanks!

Dreame L10s Ultra User here. I used the Xiaomi Home app sofar and could connect with my credentials to the Dreame Vacuum integration. 
Now my wife prefers the "original" Dreamehome App - and I can´t connect from HA.
"could not login to Xiaomi Mioo Cloud, check the credentials"

from what I´ve read above it should/might work? What am I doing wrong?

**Comment by GL640** [comment]
i have now updated to b12 ( beta ) and now nothing works .
i am going to downgrade again and see if it works again and it was the b12update?
i am using the dreame login


edit
removed b12 then rebooted 
then installed b12 again
but same problem now... strange before i updated to b12 it all worked

**Reply by Tasshack** [reply]
Can you check the logs for any possible warnings or errors?

**Comment by elenril2** [comment]
After installing the beta version and selecting the option to log in via Dreamehome App, integration during running cleaning consumes a lot of server resources. This places a very heavy load on the home assistant. I don't see any error or warning messages in the log. I don't think it's also related to spam the database by collecting historical entity states. Maybe there's a problem with the code itself?
Has anyone else noticed this behavior? I saw one similar comment on reddit - [https://www.reddit.com/r/homeassistant/comments/16xzk0w/comment/kck1qkz/](url)

**Reply by Tasshack** [reply]
Python is not suitable for generating digital images and requires too much cpu/memory and there is nothing I can do about that. 
https://www.reddit.com/r/Python/comments/ra2aqh/is_python_really_too_slow/

Also you cannot expect much performance if you are trying to run this integration on a 8 year old ARM machine like the RPI3.

You can try to enable `Low Resolution Map` and disable some map objects from configuration options to have some performance improvements.

**Reply by elenril2** [reply]
Thank you for your immediate response. My server is not running on RPI3, but the fact is that it is not a machine adapted to this type of calculations. I will try to lower the map resolution. By the way - maybe it is possible to generate it less frequently, at intervals set by the user? Maybe you could consider this option as an enhancement.

**Reply by Tasshack** [reply]
You need a new mini pc not an enhancement.
